### PR TITLE
Add support to edit item collection - admin item collection page

### PR DESCRIPTION
**Pull request by vidya-ram**
Add support to edit item collection
1. Added a reusable BaseframeForm ractive component to render forms. It uses the form template received from Baseframe. This form component can be reused in other admin pages as well.
2. This PR uses Baseframe.Forms.handleAjaxPost for post requests. (Dependency on Baseframe PR https://github.com/hasgeek/baseframe/pull/182)

### CODE REVIEW

File: boxoffice/assets/js/views/org.js [hunk 1]
```diff
@@ -45,9 +41,40 @@ export const OrgView = {
             errorMsg = formErrorHandler(response, this.get('formId'));
             orgComponent.set('icForm.errorMsg', errorMsg);
           });
+        }
+      });
+
+      let EditICForm = Ractive.extend({
```
Comment (shreyas-satish) [inline boxoffice/assets/js/views/org.js]: Why do the edit and new forms have to be separate components? What are the parts that are actually different?

File: boxoffice/assets/js/models/util.js [hunk 2]
```diff
@@ -31,6 +31,11 @@ export const Util = {
   },
   getElementId: function(htmlString) {
     return htmlString.match(/id="(.*?)"/)[1];
+  },
+  getFormTemplate: function(htmlString, submitHandler) {
```
Comment (shreyas-satish) [inline boxoffice/assets/js/models/util.js]: Isn't `registerSubmitHandler` (removed below) simpler and easier to understand? Why is this better?

File: boxoffice/assets/js/templates/org.html.js [hunk 3]
```diff
@@ -20,18 +20,29 @@ export const orgTemplate = `
...
-            <p class="section-title">Item collection id</p>
-            <p class="section-content">{{ id }}</p>
-            <p class="section-title">Item collection description</p>
-            <div class="section-content">{{{ description }}}</div>
-            <div class="btn-wrapper">
-              <a class="boxoffice-button boxoffice-button-action" href="/{{orgName}}/{{name}}">View listing</a>
-              <a class="boxoffice-button boxoffice-button-action" href="javascript:void(0)" on-click="viewDashboard(event)">View dashboard</a>
+          {{#if showEditForm}}
+            <div class="content-box clearfix" intro='fly:{"x":20,"y":"0"}'>
+              <EditICForm formTemplate="{{ formTemplate }}" ic="{{ item_collection }}" icId="{{ id }}"></EditICForm>
```
Comment (shreyas-satish) [inline boxoffice/assets/js/templates/org.html.js]: Related to the comment below, can we reuse this component for the `new` action as well?

File: boxoffice/assets/js/templates/org.html.js [hunk 4]
```diff
@@ -21,13 +21,13 @@ export const orgTemplate = `
...
           <div class="heading-edit">
-            <button class="edit-btn" on-click="showEditICForm(event)"><i class="fa fa-edit"></i>{{#loadingEditForm}}<i class="fa fa-spinner fa-spin">{{/}}</button>
+            <button class="edit-btn" on-click="showIcForm(event, 'edit')"><i class="fa fa-edit"></i>{{#loadingEditForm}}<i class="fa fa-spinner fa-spin">{{/}}</button>
           </div>
         </div>
         <div class="content">
           {{#if showEditForm}}
             <div class="content-box clearfix" intro='fly:{"x":20,"y":"0"}'>
-              <EditICForm formTemplate="{{ formTemplate }}" ic="{{ item_collection }}" icId="{{ id }}"></EditICForm>
+              <ICForm formTemplate="{{ formTemplate }}" ic="{{ item_collection }}" icId="{{ id }}"></ICForm>
```
Comment (shreyas-satish) [inline boxoffice/assets/js/templates/org.html.js]: Can we abstract this further and make it `Form`? That way we can reuse this across models?

File: boxoffice/assets/js/views/org.js [hunk 5]
```diff
@@ -85,36 +73,36 @@ export const OrgView = {
...
           showAddForm: DEFAULT.hideForm,
-          editIcForm: ''
         },
-        components: {ICForm: ICForm, EditICForm: EditICForm},
-        showNewIcForm: function (event) {
-          this.set('showAddForm', DEFAULT.showForm);
+        components: {ICForm: ICForm},
+        showIcForm: function (event, action) {
+          if (action === "edit") {
+            let ic = event.keypath,
```
Comment (shreyas-satish) [inline boxoffice/assets/js/views/org.js]: Can you help explain what's happening here, I'm a little confused.

Comment (vidya-ram) [reply boxoffice/assets/js/views/org.js]: In case the action is `edit` Item collection, client does an ajax call to fetch the item collection edit form.

**Comment by shreyas-satish** [comment]
Addressed in https://github.com/hasgeek/boxoffice/pull/211

**Pull request closed without merging**